src/language-css/parser-postcss.js

```diff
@@ -17,11 +17,8 @@ function parseSelector(selector) {
     };
   }
   const selectorParser = require("postcss-selector-parser");
-  let result;
-  selectorParser(result_ => {
-    result = result_;
-  }).process(selector);
-  return addTypePrefix(result, "selector-");
+  const ast = selectorParser().astSync(selector);
+  return addTypePrefix(ast, "selector-");
```

Comment:
Mutating our value for `type` and removing the `parent` back-reference is unsupported and may cause bugs in future minor releases. I recommend that you stop doing this.

If you want to associate your own data with our nodes you can set your own attributes but this is likely to force v8 to de-opt (the delete on the `parent` property is likely already doing this -- what is the point of that?). A good solution is to use a WeakMap to set your own values to be associated with any of our nodes. these will get GC'd automatically.

Reply:
I don't understand, what exactly is the problem doing this? Once we get the AST, what exactly should be "supported"?

We're only interested in traversing the tree top-down and have our own structure to traverse it (and get the parents, see: https://github.com/prettier/prettier/blob/master/src/common/fast-path.js)

Reply:
Given that it works, there is no problem at this time. But removing the parent has certainly introduced bugs in methods that you are not using. E.g. `node.replace()`. Changing type has broken our [typeGuard methods](https://github.com/postcss/postcss-selector-parser/blob/master/src/selectors/guards.js).

I'm just letting you know that it's unsupported and I can't guarantee patch release compatibility when you mutate internals like this. Please consider how you can structure this code so that such changes to our internals are not performed. 🙏

Reply:
Understood!

What I meant is, we treat the output of the parser as a simple object and don't intentionally use any "methods" of it.

Reply:
The AST for selectors has multiple properties that are used in coordination to handle a variety of edge cases. This approach will invariably lead to you implementing your own version of our methods or having bugs that are hard to catch due to incomplete test cases. 

In the latest version we handle escapes transparently for you. This is a semantic change that very likely fixes a number of bugs you didn’t know you had but if you don’t read or set the properties correctly you could result in changing the original source in a way that you don’t want.

Are there test cases in prettier around various escapes and extended Unicode characters in selectors? If so can you point me to them so I can verify that they are testing aspects that are especially tricky to get right?